mr/src/main/java/org/apache/iceberg/mr/mapred/serde/objectinspector/IcebergBinaryObjectInspector.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.mapred.serde.objectinspector;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.BinaryObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.hadoop.io.BytesWritable;
+
+public final class IcebergBinaryObjectInspector extends IcebergPrimitiveObjectInspector
+                                                implements BinaryObjectInspector {
+
+  private static final IcebergBinaryObjectInspector INSTANCE = new IcebergBinaryObjectInspector();
+
+  public static IcebergBinaryObjectInspector get() {
+    return INSTANCE;
+  }
+
+  private IcebergBinaryObjectInspector() {
+    super(TypeInfoFactory.binaryTypeInfo);
+  }
+
+  @Override
+  public byte[] getPrimitiveJavaObject(Object o) {
+    return o == null ? null : ((ByteBuffer) o).array();
```

Comment:
This isn't correct because it doesn't follow the contract of `ByteBuffer`. Avro will reuse byte buffers, so there is no guarantee that this array is the correct length. In addition, we want to generally follow the `ByteBuffer` contract so that we don't need to worry about whether an optimization later (buffer reuse) will break certain sections of code.

An easy fix is to use `ByteBuffers.toByteArray` here.